### ISSUE TITLE
Conditionally allow None

### ISSUE THREAD

**Issue by jamesob** [kislyuk/ensure#13]
Often when instantiating classes with kwargs, you want to conditionally validate a parameter's type if it isn't `None`, e.g.

```python
class SomeClass(object):
    def __init__(self, a=None):
        # assert `a` is an int if it isn't None
```

It'd be nice to facilitate straightforward expression of this with `ensure`. Maybe something like

```
check(a).is_none_or_an(int)
```
or maybe
```
check(a).is_an(int, allow_none=True)
```

**Comment by kislyuk** [comment]
Sorry about the lengthy delay. This is a cool idea. I'd like to implement it, but in the meantime, PRs are welcome. I'd like to avoid duplicating all function names to add the `is_none_or_`, if possible.

**Comment by kislyuk** [comment]
Actually, because this is always going to be used as a predicate, it can be introduced through an attribute access. In other words, if the attribute `is_none_or` is accessed, configure the inspector to accept None, and return self. Then you can say things like `ensure(a).is_none_or.is_an(int)`. So far I don't see anything horribly broken with this option.

I'm less thrilled about the kwarg option, but I'm not sure that there's anything wrong with it either.

**Comment by kislyuk** [comment]
Released in v0.3.3.

**Comment by jamesob** [comment]
Nice, thanks @kislyuk
